### PR TITLE
The documentation of most of the files have been updated according to my knowledge and understanding of wplay

### PR DESCRIPTION
**Pull request by 07ATUL**
@xandao6 @rpotter12 please review and merge.

### CODE REVIEW

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 6
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5140051&bid=16886238)

File: wplay/onlinetracker.py [hunk 1]
```diff
@@ -1,24 +1,35 @@
...
 from wplay.utils import target_search
 from wplay.utils import target_data
 from wplay.utils.helpers import data_folder_path
 
-
+# After the client run the command line script, the user enters a name and that is the target which is used here.
 async def tracker(target):
+    # Waiting for Whatsapp web to be loading
     page, _ = await browser_config.configure_browser_and_load_whatsapp()
+    # Searching for the target name
```
Comment (rpotter12) [inline wplay/onlinetracker.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5140051)

File: wplay/onlinetracker.py [hunk 2]
```diff
@@ -1,24 +1,35 @@
...
     status_file = open(data_folder_path / 'tracking_data' / f'status_{target_name}.txt', 'a')
 
     is_sound_enabled = True
+    
     last_status = 'offline'
     try:
+        # After username is found it prints the various name if matching strings are there and asks the user for which name to track and sets that name to target.
         print(f'Tracking: {target_name}')
+        # It writes the status whether offline or online in tracking file.
+
```
Comment (rpotter12) [inline wplay/onlinetracker.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5140051)

File: wplay/wchat.py [hunk 3]
```diff
@@ -1,13 +1,15 @@
...
 from wplay.utils import io
 
-
+# First checks for the target which is the username as entered by the client after running wplay on a command line script
 async def chat(target):
     page, _ = await browser_config.configure_browser_and_load_whatsapp()
 
     await target_search.search_and_select_target(page, target)
-
+
```
Comment (rpotter12) [inline wplay/wchat.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5140051)

File: wplay/utils/session_manager.py [hunk 4]
```diff
@@ -1,4 +1,4 @@
-# region Imports
+# region Imports
```
Comment (rpotter12) [inline wplay/utils/session_manager.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5140051)

File: wplay/messageblast.py [hunk 5]
```diff
@@ -1,12 +1,21 @@
...
 
 async def blast(target):
     page, _ = await browser_config.configure_browser_and_load_whatsapp()
+    # It searches the username by putting the name in search bar by using divison selectors and if more username found asks for your target.
     await target_search.search_and_select_target(page, target)
+    
     message = io.ask_user_for_message_breakline_mode()
+    
     number_of_messages = int(input("Enter the number of messages to blast: "))
+
```
Comment (rpotter12) [inline wplay/messageblast.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5140051)

File: setup.py [hunk 6]
```diff
@@ -19,14 +24,22 @@
...
+    # all about the software description
+    
     long_description_content_type="text/markdown",
     author="Rohit Potter, Alexandre Calil",
     author_email="rohitpotter12@gmail.com, alexandrecalilmf@gmail.com",
     license="MIT",
+    
+    #The minimum python version requirement
     python_requires=">=3.6",
+
```
Comment (rpotter12) [inline setup.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5140051)

**Comment by 07ATUL** [comment]
@rpotter12 Can you please guide me how to resolve it

File: wplay/messageblast.py [hunk 7]
```diff
@@ -1,11 +1,16 @@
...
 from wplay.utils import io
-
-
+# This function gets the input which is username as given by the client and checks for chromedriver to load Whatsapp web.
 async def blast(target):
     page, _ = await browser_config.configure_browser_and_load_whatsapp()
+# It searches the username by putting the name in search bar by using divison selectors and if more username found asks for your target.
+    await target_search.search_and_select_target(page, target)
+    message = io.ask_user_for_message_breakline_mode() 
+    number_of_messages = int(input("Enter the number of messages to blast: "))
```
Comment (rpotter12) [inline wplay/messageblast.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5140051)

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 7
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5140051&bid=16886238)

File: wplay/utils/session_manager.py [hunk 8]
```diff
@@ -82,8 +76,8 @@ def prepare_questions(self):
                 'choices': list(map(lambda e: {'name': e}, self.data_filenames)),
                 'when': lambda answers: answers['user_options'] == self.user_options['delete']
             }
-        ]
-
+        ]
```
Comment (rpotter12) [inline wplay/utils/session_manager.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5140051)

File: wplay/onlinetracker.py [hunk 9]
```diff
@@ -1,28 +1,35 @@
...
 from wplay.utils import target_data
 from wplay.utils.helpers import data_folder_path
-
-
+# After the client run the command line script, the user enters a name and that is the target which is used here.
 async def tracker(target):
+# Waiting for Whatsapp web to be loading
     page, _ = await browser_config.configure_browser_and_load_whatsapp()
+
+# Searching for the target name
```
Comment (rpotter12) [inline wplay/onlinetracker.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5140051)

File: wplay/onlinetracker.py [hunk 10]
```diff
@@ -1,28 +1,35 @@
...
     status_file = open(data_folder_path / 'tracking_data' / f'status_{target_name}.txt', 'w').close()
     status_file = open(data_folder_path / 'tracking_data' / f'status_{target_name}.txt', 'a')
-
-    is_sound_enabled = True
+    is_sound_enabled = True 
     last_status = 'offline'
     try:
+# After username is found it prints the various name if matching strings are there and asks the user for which name to track and sets that name to target.
         print(f'Tracking: {target_name}')
+# It writes the status whether offline or online in tracking file.
```
Comment (rpotter12) [inline wplay/onlinetracker.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5140051)

File: wplay/onlinetracker.py [hunk 11]
```diff
@@ -1,28 +1,35 @@
...
     if target is not None:
         target_name = await target_search.search_and_select_target(page, target, hide_groups = True)
     else:
         target_name = await target_select.manual_select_target(page, hide_groups = True)
     Path(data_folder_path / 'tracking_data').mkdir(parents = True, exist_ok = True)
     status_file = open(data_folder_path / 'tracking_data' / f'status_{target_name}.txt', 'w').close()
     status_file = open(data_folder_path / 'tracking_data' / f'status_{target_name}.txt', 'a')
-
-    is_sound_enabled = True
+    is_sound_enabled = True
```
Comment (rpotter12) [inline wplay/onlinetracker.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5140051)

File: wplay/messageblast.py [hunk 12]
```diff
@@ -1,11 +1,16 @@
...
 from wplay.utils import target_select
 from wplay.utils import io
-
-
+# This function gets the input which is username as given by the client and checks for chromedriver to load Whatsapp web.
 async def blast(target):
     page, _ = await browser_config.configure_browser_and_load_whatsapp()
+# It searches the username by putting the name in search bar by using divison selectors and if more username found asks for your target.
+    await target_search.search_and_select_target(page, target)
+    message = io.ask_user_for_message_breakline_mode()
```
Comment (rpotter12) [inline wplay/messageblast.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5140051)

File: wplay/utils/session_manager.py [hunk 13]
```diff
@@ -1,20 +1,17 @@
-# region Imports
+# region Imports
```
Comment (rpotter12) [inline wplay/utils/session_manager.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5140051)

File: wplay/__main__.py [hunk 14]
```diff
@@ -1,24 +1,24 @@
+# The first four lines import python library scripts.
```
Comment (rpotter12) [inline wplay/__main__.py]: No need of this line

File: wplay/__main__.py [hunk 15]
```diff
@@ -29,26 +29,22 @@ def get_arg_parser():
         default=None,
         nargs="?",
         help="contact or group name, optional , target can be selected manually except for saving chat")
-
-    group = parser.add_mutually_exclusive_group(required = True)
+    group = https://github.com/rpotter12/whatsapp-play/pull/199/conflict?name=wplay%252Fonlinetracker.py&ancestor_oid=098a895dcee0146565d3614cc50e82da44c4d532&base_oid=2c03f17cffb9deaf4b14def1b267e9d62ff1953f&head_oid=120a111cc8fb7771469b8e64b9f62d962c0146ffparser.add_mutually_exclusive_group(required = True)
```
Comment (rpotter12) [inline wplay/__main__.py]: Why you changed this?? Please explain.

File: wplay/messagetimer.py [hunk 16]
```diff
@@ -1,14 +1,17 @@
+# First two lines import python libraries
```
Comment (rpotter12) [inline wplay/messagetimer.py]: No need of this

File: wplay/messagetimer.py [hunk 17]
```diff
@@ -35,3 +39,4 @@
             time.sleep(random.randrange(minimumTimeInterval, maximumTimeInterval))
         else:
             time.sleep(minimumTimeInterval)
+# This script is a helper script for message blast script as the no of messages and the message time are defined here and are being using in the blast script.
```
Comment (rpotter12) [inline wplay/messagetimer.py]: script explanation is always in the starting

**Comment by xandao6** [comment]
Also do a rebase please, I've changed setup and requirements

**Comment by xandao6** [comment]
@07ATUL

**Comment by rpotter12** [comment]
@07ATUL please rebase your fork. :)

**Comment by 07ATUL** [comment]
> @07ATUL please rebase your fork. :)

sir how to do it please guide

**Comment by priyansh19** [comment]
https://gist.github.com/ravibhure/a7e0918ff4937c9ea1c456698dcd58aa refer this link .. for using rebase command :+1:

**Comment by rpotter12** [comment]
@07ATUL Please rebase your fork :)

**Comment by rpotter12** [comment]
@07ATUL Mention issue number also in this PR

**Pull request closed without merging**